common/src/main/java/net/opentsdb/data/TimeSeriesId.java

```diff
@@ -0,0 +1,122 @@
+   * storage engine.
+   */
+  public boolean encoded();
+  
+  /**
+   * A simple string for identifying the time series. The alias may be null or
+   * empty. If a value is present, the alias must be unique within a set of
+   * time series.
+   * <p>
+   * Invariant: If all other fields for the identity are empty, then the alias
+   * must be non-null and non-empty.
+   * 
+   * @return A string or null if not used.
+   */
+  public String alias();
+  
+  /**
+   * A list of namespace identifiers included in the sources for this time series.
+   * May be empty if namespaces are not in use for the platform.
+   *  
+   * @return A non-null list if zero or more namespace identifiers.
+   */
+  public List<String> namespaces();
+  
+  /**
+   * A list of metric names included in the sources for this time series.
+   * If this list is empty, check {@link #metrics()} for data.
+   * <p>
+   * Invariant: If this list is empty (meaning the series or set of series is 
+   * not associated with a specific metric) then the {@link #alias()} must be 
+   * non-null and unique.
+   *  
+   * @return A non-null list of zero or more metric names.
+   */
+  public List<String> metrics();
+  
+  /**
+   * A list of tag name and value pairs included in the sources for this time 
+   * series. If the underlying storage system does not support tags or none of
+   * the source time series had tags pairs in common, this list may be empty.
+   * The map key represents a tag name (tagk) and the map value represents a 
+   * tag value (tagv).
+   * <p>
+   * Invariant: Each tag pair must appear in every source time series.
+   * 
+   * @return A non-null map of zero or more tag pairs.
+   */
+  public Map<String, String> tags();
+
+  /**
+   * A list of tag names (tagk) that were represented in every source series
+   * but had one or more differing tag values (tagv). This list may be empty if
+   * - The underlying store does not support tags.
+   * - All of the source time series shared the same tag pairs.
+   * - All of the source time series had disjoint tag sets.
+   * <p>
+   * Invariant: Each tag name must be present in every time series if it appears
+   * in this list.
+   * 
+   * @return A non-null list of zero or more tag names.
+   */
+  public List<String> aggregatedTags();
+  
+  /**
+   * A list of tag names (tagk) that were represented in one or more source time
+   * series but not every source time series. E.g. series A has tag name Z but
+   * series B does not.
+   * This list may be empty if:
+   * - The underlying store does not support tags.
+   * - All of the source time series shared the same tag pairs.
+   * - All of the source time series shared the same tag names with different
+   * values.
+   * <p>
+   * Invariant: Each tag name must be present in at least one but not all 
+   * source series if it appears in this list.
+   * 
+   * @return A non-null list of zero or more tag names.
+   */
+  public List<String> disjointTags();
```

Comment:
Can we please document how disjoint tags are useful.

Reply:
aye